### ISSUE TITLE
MemoryError when running "Evaluate all standard scenarios" in ArcMap

### ISSUE THREAD

**Issue by HEdingfield** [Geosyntec/python-tidegates#46]
This assumes PR #45 is merged in. Traceback:

    Messages
    Executing: StandardScenarios C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb\dem_x04 C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb\ZOI GeoID output_evaluate # # # #
    Start Time: Tue Nov 17 15:28:31 2015
    Running script StandardScenarios...

    Analyzing flood elevation: 4.0 ft (MHHW, 0)
    -------------------------------------------
    ... (this ran fine) ...

    Analyzing flood elevation: 5.0 ft (MHHW, 1)
    -------------------------------------------
    ... (this ran fine) ...

    Analyzing flood elevation: 6.0 ft (MHHW, 2)
    -------------------------------------------
    WorkSpace set to C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb
    Loading DEM C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb\dem_x04
    Processing C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb\ZOI polygons
    Clipping DEM to extent of C:\Users\HEdingfield.GEOSYNTEC\Desktop\temp\MB_Cape.gdb\ZOI
    Converting rasters to arrays
    Flooding areas up to 6.0 ft

    Traceback (most recent call last):
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\toolbox.py", line 177, in execute
        self.main_execute(**params)
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\toolbox.py", line 780, in main_execute
        **params
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\toolbox.py", line 662, in analyze
        asMessage=True
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\analysis.py", line 130, in flood_area
        **verbose_options
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\utils.py", line 285, in wrapper
        return func(*args, **kwargs)
      File "H:\Dev\PyCharm Projects\python-tidegates\tidegates\utils.py", line 813, in flood_zones
        flooded_array = zones_array.copy()
    MemoryError

    Failed to execute (StandardScenarios).
    Failed at Tue Nov 17 15:30:23 2015 (Elapsed Time: 1 minutes 51 seconds)

**Comment by phobson** [comment]
Turn on background geoprocessing to enable 64-bit support and report back.

![image](https://cloud.githubusercontent.com/assets/1163939/11228784/22fd2fde-8d44-11e5-9064-5f2a6df95b4d.png)

**Comment by HEdingfield** [comment]
I already had this enabled exactly as your screenshot shows. I just ran it again though and it made it to 7.0 ft. but still crapped out after "Converting rasters to arrays" with a MemoryError.

**Comment by phobson** [comment]
closing this as its a hardware issue (for the moment).

**Comment by HEdingfield** [comment]
After installing 64-bit background processing, I finally stopped hitting the MemoryError. However, my computer hard froze (after I assumed it ran out of RAM) when I tried running it the first time about 1/3 of the way through. I rebooted and closed *everything* but Arc, and was able to successfully run all standard scenarios. However, it was extremely taxing on my 8 GB of RAM.

I strongly recommend noting in the docs that 64-bit background processing (i.e. the 64-bit version of Python 2.7 for ArcGIS) is required, and at least 8 GB of RAM needs to be available to actually run the scenarios.
